homeassistant/components/file/__init__.py

```diff
@@ -1 +1,30 @@
 """The file component."""
+
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+from homeassistant.helpers.typing import ConfigType
+
+DOMAIN = "file"
+
+WRITE_PARAM_FILENAME = "filename"
+WRITE_PARAM_CONTENT = "content"
+WRITE_PARAM_MODE = "mode"
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the file write service."""
+
+    @callback
+    def write(call: ServiceCall) -> None:
+        """Write a file."""
+        filename = call.data[WRITE_PARAM_FILENAME]
+        content = call.data[WRITE_PARAM_CONTENT]
+        mode = call.data.get(WRITE_PARAM_MODE, "w")
+
```

Comment:
You probably need to check `hass.config.is_allowed_path(path)` here

Reply:
Hmm... I'm trying to understand `is_allowed_path` -- it looks like it's checking [this](https://www.home-assistant.io/docs/configuration/basic/#allowlist_external_dirs) value, but it's not obvious to me that this configuration parameter is meant for whether one can write to the directory (vs. read from it).  Maybe there should just be a list of allowed directories or file patterns in the config for the integration?

Reply:
For now I went ahead and added the allowlist and am checking against it.  Feel free to unresolve if you'd prefer I do things a different way.